tests/connector_test.go

```diff
@@ -28,67 +24,29 @@ var inOutConfigGitPullIfNotPresent = `
 `
 
 func TestRunStepGit(t *testing.T) {
-	moduleName := "arcaflow-plugin-example@git+https://github.com/arcalot/arcaflow-plugin-example.git"
-	connector, _ := getConnector(t, inOutConfigGitPullAlways, nil)
-	OutputID, OutputData, Error := RunStep(t, connector, moduleName)
-	assert.NoError(t, Error)
-	assert.Equals(t, OutputID, "success")
-	assert.Equals(t,
-		OutputData.(map[interface{}]interface{}),
-		map[interface{}]interface{}{"message": fmt.Sprintf("Hello, %s!", examplePluginNickname)})
-}
-
-func TestPullPolicies(t *testing.T) {
-	moduleName := "arcaflow-plugin-example@git+https://github.com/arcalot/arcaflow-plugin-example.git"
-	// this test must be run in the same workdir so it's created upfront
-	// and passed to the getConnector func
-	workdir := createWorkdir(t)
-	connectorAlways, _ := getConnector(t, inOutConfigGitPullAlways, &workdir)
-	connectorIfNotPresent, _ := getConnector(t, inOutConfigGitPullIfNotPresent, &workdir)
-	// pull mode Always, venv will be removed if present and pulled again
-	OutputID, OutputData, Error := RunStep(t, connectorAlways, moduleName)
-	assert.NoError(t, Error)
-	assert.Equals(t, OutputID, "success")
-	assert.Equals(t,
-		OutputData.(map[interface{}]interface{}),
-		map[interface{}]interface{}{"message": fmt.Sprintf("Hello, %s!", examplePluginNickname)})
-	// pull mode IfNotPresent, venv will be kept
-	OutputID, OutputData, Error = RunStep(t, connectorIfNotPresent, moduleName)
-	assert.NoError(t, Error)
-	assert.Equals(t, OutputID, "success")
-	assert.Equals(t,
-		OutputData.(map[interface{}]interface{}),
-		map[interface{}]interface{}{"message": fmt.Sprintf("Hello, %s!", examplePluginNickname)})
-	wrapper := getCliWrapper(t, workdir)
-	path, err := wrapper.GetModulePath(moduleName)
-	assert.NoError(t, err)
-	file, err := os.Stat(*path)
-	assert.NoError(t, err)
-	// venv path modification time is checked
-	startTime := file.ModTime()
-	// pull mode Always, venv will be removed if present and pulled again
-	OutputID, OutputData, Error = RunStep(t, connectorAlways, moduleName)
+	//moduleName := "arcaflow-plugin-example@git+https://github.com/arcalot/arcaflow-plugin-example.git"
+	moduleName := "arcaflow-plugin-template-python@git+https://github.com/arcalot/arcaflow-plugin-template-python.git@1dd9a772e65397f5f56af3b907f84fcdfbd9ff28"
```

Comment:
I see embedding the commit reference as pinning the version of the plugin repo. As long as the reference exists, then the test will not break because of a change in the plugin repo, and we only care that the plugin repo is an executable python module.

We could postpone this PR to make the changes to those plugins and add tags or release versions and pin the test to use those, but that also seems worth a separate issue to me.

Reply:
If the test doesn't care which version of the plug-in that it pulls, then it would be fine to pin it to a particular one...but, then again, it would be fine to use `latest` (or to omit the hash, right?), and using `latest` would continue to work even if the target repo were changed or purged or recreated without our favorite commit in it.

On the other hand, if the version matters, then it almost certainly shouldn't be pinned to a hash, because that creates a future technical debt, not to mention a possible silent failure of the test (i.e., when the plugin is updated but this test doesn't test the new version because it is pinned to the old one).

If you really want it pinned, then you should probably be using a symbolic reference here which is defined somewhere else (e.g., in the test harness, independent of the test source), so that it can be managed as part of the test configuration.